src/map/map.cpp

```diff
@@ -3558,6 +3558,9 @@ int map_readfromcache(struct map_data *m, char *buffer, char *decode_buffer)
 	for(i = 0; i < header->map_count; i++) {
 		info = (struct map_cache_map_info *)p;
 
+		// name exists in maps_athena.conf but not in map_cache.dat
+		if (info->name != nullptr && info->name[0] == '\0')
```

Comment:
wouldn't be simplier to have this check inverted ?
`if(info->name == nullptr || info->name[0] == '\0')`
I mean, while we are checking for nullptr we might as well quit if it's the case. The previous check would have allow a nullptr.